sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/Statistics.scala

```diff
@@ -119,6 +120,18 @@ case class ColumnStat(
       maxLen = maxLen,
       histogram = histogram,
       version = version)
+
+  def updateCountStats(
+      oldNumRows: BigInt,
+      newNumRows: BigInt,
+      updatedColumnStatOpt: Option[ColumnStat] = None): ColumnStat = {
```

Comment:
this parameter looks a bit weird, I'd expect the caller side to do
`updatedMap.get(attr.exprId).map(_._2).getOrElse(oriColStat).updateStates(oldNumRows, newNumRows)`

Reply:
I thought that would work too, but turns out that we actually need `oriColStat` to avoid updating the "updated" statistic.

Reply:
Ah I see the related code
```
      else if (colStat.distinctCount.get > 1) {
        // Update ndv based on the overall filter selectivity: scale down ndv if the number of rows
        // decreases; otherwise keep it unchanged.
        Some(EstimationUtils.updateNdv(oldNumRows = rowsBeforeFilter,
          newNumRows = rowsAfterFilter, oldNdv = oriColStat.distinctCount.get))
```